File: 프로그래머스/unrated/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int[] answer = new int[2];
        if(denom1 == denom2) {
            answer[0] = numer1 + numer2;
            answer[1] = denom1;
        }
        else {
            answer[0] = numer1 * denom2 + denom1 * numer2;
            answer[1] = denom1 * denom2;
        }
        
        int num = 2;
        while(true){
            if(answer[0] % num == 0 && answer[1] % num == 0) {
                answer[0] /= num;
                answer[1] /= num;
                num--;
            }
            num++;
            
            if(num > answer[0] && num > answer[1]) break;
        }
        return answer;
    }
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.py
import math

def solution(numer1, denom1, numer2, denom2):
    fir = numer1 * denom2 + numer2 * denom1
    sec = denom1 * denom2
    return [fir/math.gcd(fir, sec), sec/math.gcd(fir, sec)]